source/onboard/migrate-from-slack.rst

```diff
@@ -0,0 +1,191 @@
+
+The migration is idempotent, meaning that you can run multiple imports that contain the same posts, and there will not be duplicated created posts in Mattermost. Each post is imported with the correct user/author and ``created_at`` value from your Slack instance. Threads are kept in tact with the import.
+
+Option 1: Upload export via ``mmctl``
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Ensure you have the Mattermost command line tool ``mmctl`` installed. This allows you to perform different tasks that communicate to Mattermost's API. You'll also want to `configure authentication </manage/mmctl-command-line-tool.html#mmctl-auth>`__ for the tool.
+
+To prepare our files to be uploaded to the server, we need to put both the ``.jsonl`` file and ``data`` folder together into a zip file.
+
+.. code:: bash
+
+  zip -r mattermost-bulk-import.zip data mattermost_import.jsonl
+
+Then we can upload the zip file to our Mattermost server:
+
+.. code:: bash
+
+  mmctl import upload ./mattermost-bulk-import.zip
+
+Run this command to list the available imports:
+
+.. code:: bash
+
+  mmctl import list available
+
+Run this command to process the import. Replace ``<IMPORT FILE NAME>`` with the name you got from the ``mmctl import list available`` command:
+
+.. code:: bash
+
+  mmctl import process <IMPORT FILE NAME>
+
+Finally, run this command to view the status of the import process job. If the job status shows as ``pending``, then wait before running the command again. The ``--json`` flag is required to view the possible error message. Replace ``<JOB ID>`` with the id you got from the ``mmctl import list process`` command:
+
+.. code:: bash
+
+  mmctl import job show <JOB ID> --json
+
+Option 2: Use the ``mattermost`` command to run the export
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+First get the ``.jsonl`` file and ``data`` folder onto your server using something like ``scp``. Then you'll use the ``mattermost`` binary to process the bulk import data:
+
+.. code:: bash
+
+    mattermost import bulk ./mattermost_import.jsonl --import-path ./data --apply
+
+Debug imports
+^^^^^^^^^^^^^
+
+If you choose the ``mmctl`` option above, you can use the ``mmctl import job show`` command to view any relevant errors that may have occurred. If you run into problems which the error message does not help to resolve, please try the first option of using the ``mattermost bulk import`` command. The ``mmctl`` import process does not give you any additional debugging information, even in the Mattermost server logs.
+
+.. important::
+
+    Note that if your logged in Mattermost user is not present in your input (neither the email nor the username matches any users in the import file), your user will not be added to any new channels in the import process.
```

Comment:
What situation are we getting at here?  Are we kinda saying "Hey, if you are part of the user group being imported from Slack your Mattermost profile must have a matching username and email to the corresponding `User` line of the `jsonl` file.  You can manually edit the file to ensure make it match your Mattermost user"